cls/SourceControl/Git/PullEventHandler/IncrementalLoad.cls

```diff
@@ -14,12 +14,13 @@ Method OnPull() As %Status
 
     for i=1:1:$get(..ModifiedFiles){
         set internalName = ..ModifiedFiles(i).internalName
-        if ((internalName = "") && (..ModifiedFiles(i).changeType '= "D")) {
+        if (((internalName = "") && (..ModifiedFiles(i).changeType '= "D")) || (..ModifiedFiles(i).changeType = "D")) {
```

Comment:
Tweak here: if changeType is D, we need to delete the item in the database (to *really* handle file deletion)

Reply:
Note: this entails calling $System.OBJ.Delete for a class, $System.OBJ.DeletePage for a CSP file, and %Library.RoutineMgr:Delete otherwise. Would probably be worth adding a helper method for that, if we don't already have one.